packages/ui/v2/core/apps/AllApps.tsx

```diff
@@ -0,0 +1,109 @@
+import { useAutoAnimate } from "@formkit/auto-animate/react";
+import type { Credential } from "@prisma/client";
+import { UIEvent, useEffect, useRef, useState } from "react";
+import { ChevronLeft, ChevronRight } from "react-feather";
+
+import { classNames } from "@calcom/lib";
+import { useLocale } from "@calcom/lib/hooks/useLocale";
+import type { App } from "@calcom/types/App";
+
+import AppCard from "./AppCard";
+
+type AllAppsPropsType = { apps: (App & { credentials: Credential[] | undefined })[] };
+
+export default function AllApps({ apps }: AllAppsPropsType) {
+  const { t } = useLocale();
+  const [selectedCategory, setSelectedCategory] = useState<string | null>(null);
```

Comment:
This should come from the router and use shallow routing

Reply:
@zomars Wouldn't this be polluting the URL address without really a use case? 🤔

Reply:
Use case: Refreshing loses the selected category.